dbms/src/Debug/MockStorage.cpp

```diff
@@ -141,11 +146,23 @@ ColumnsWithTypeAndName MockStorage::getColumnsForMPPTableScan(Int64 table_id, In
         ColumnsWithTypeAndName res;
         for (const auto & column_with_type_and_name : columns_with_type_and_name)
         {
-            res.push_back(
-                ColumnWithTypeAndName(
-                    column_with_type_and_name.column->cut(cut_info.first, cut_info.second),
-                    column_with_type_and_name.type,
-                    column_with_type_and_name.name));
+            bool contains = false;
+            for (const auto & column : table_scan.getColumns())
+            {
+                if (column.id == column_with_type_and_name.column_id)
```

Comment:
why not use column name for comparison here?

Reply:
> why not use column name for comparison here?

Because columns in TiDBTableScan don't have column name.